harness/determined/layers/_workload_sequencer.py

```diff
@@ -165,7 +165,9 @@ def as_batches(
             return max(records // self.global_batch_size, 1)
         if epochs is not None:
             check.is_instance(self.records_per_epoch, int, "length must be an integer")
-            assert self.records_per_epoch is not None
+            assert (
+                self.records_per_epoch is not None
+            ), "records_per_epoch must be specified for epoch units"
```

Comment:
Can you elaborate a lot more on this message?  Which field did the user specify in epcohs that would cause this check to fail?  (this might not be the right place for the check, btw).

Side note, don't use `assert` to verify user input, since users can run `python -O` to ignore all assert statements.  So assertions are fine for mypy or detecting internal assumptions, but user validation must always be `if` statements.